### PR TITLE
Add promise error type

### PR DESCRIPTION
**Pull request by Sleitnick**
Promise type definition was missing information on Error.

### CODE REVIEW

File: types/Promise.d.ts [hunk 1]
```diff
@@ -4,6 +4,41 @@
...
+
+interface PromiseErrorConstructor {
+	readonly Kind: {
+		readonly ExecutionError: "ExecutionError";
+		readonly AlreadyCancelled: "AlreadyCancelled";
+		readonly NotResolvedInTime: "NotResolvedInTime";
+		readonly TimedOut: "TimedOut";
+	};
+
+	is(anything: any): boolean;
```
Comment (osyrisrblx) [inline types/Promise.d.ts:35]:
```suggestion
	is(value: unknown): value is PromiseError;
```

File: types/Promise.d.ts [hunk 2]
```diff
@@ -4,6 +4,41 @@
...
+
+	isKind(anything: any, kind: keyof PromiseErrorConstructor["Kind"]): boolean;
```
Comment (osyrisrblx) [inline types/Promise.d.ts:37]: This could probably also be another `: value is PromiseError`?

File: types/Promise.d.ts [hunk 3]
```diff
@@ -4,6 +4,41 @@
 // Based on roblox-lua-promise v3.1.0
 // https://eryn.io/roblox-lua-promise/
 
+interface PromiseErrorOptions {
+	trace: string;
+	context: string;
+	kind: keyof PromiseErrorConstructor["Kind"];
```
Comment (osyrisrblx) [inline types/Promise.d.ts:10]: I think we could make this `kind: Promise.Error.Kind;` with a bit of namespace wrangling?

**Review by osyrisrblx — CHANGES_REQUESTED** [review]

File: types/Promise.d.ts (hunk 1 above)
Comment (Sleitnick) [reply types/Promise.d.ts:35]: Oh that's interesting, I've never seen that syntax before for a return type

Comment (osyrisrblx) [reply types/Promise.d.ts:35]: These are called "type predicates" and are how all of the `@rbxts/t` types work!

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

**Comment by jackTabsCode** [comment]
Bumping this, trying to detect if a Promise errored due to a timeout and can't!

**Pull request merged**